### DISCUSSION TITLE
Release 2.1.0

### DISCUSSION THREAD

**Discussion by IsaacMarovitz** [Whisky-App/Whisky#506] [Announcements]
**New Features:**
- Added [**MSync**](https://github.com/marzent/wine-msync), a Mac-specific implementation of Wine sync, which greatly improves FPS in CPU-bound applications (GPTK needs to be re-downloaded) 

**Bug Fixes:**
- Added a minimum window size to Whisky's main window
- Fixed programs starting from pins not using arguments and environment variables
- Fixed an issue where Retina Mode toggle would fail on other languages if the key didn't exist

**Full Changelog:** https://github.com/Whisky-App/Whisky/compare/v2.0.2...v2.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Whisky-App/Whisky/releases/tag/v2.1.0'>Release 2.1.0</a>.</em>

**Comment by ghost** [comment]
Definitely noticing less slowdown in Portal 2 and Pixeljunk Shooter, very cool, thank you!

**Comment by clearluyy** [comment]
With MSync,grand theft auto V gets 75fps(15 higher than older version) pretty cool!

**Comment by aquaibm** [comment]
After upgrading from 2.0.2 to 2.1.0, Diablo IV is no longer loading, even though it was working fine before. On the Diablo IV page on battle.net, when I click the play button, it only displays ‘Playing Now’ for a few seconds, then changes to ‘Update’. Once the ‘Update’ is complete and it displays ‘Play’, I click ‘Play’, but it just repeats the previous steps and Diablo IV never loads.

**Reply by larryisthere** [reply]
Diablo II: Resurrected can't be played either. After clicking "Play," only a black window appears.

**Reply by OldTabdy** [reply]
Yes, the same here, the game goes to a black screen. I did a clean install of 2.1, and nothing changed.

**Reply by Yasheed1995** [reply]
I cannot play D2R either...

--update
after switing back to v2.0.2, I can play again

**Reply by ohaiibuzzle** [reply]
For those of you with this issue, turn off MSync, then create an environment variable called "WINEESYNC" with value of `1` to revert to ESync.
The issue is suspected to be that these games expect a small delay but MSync is way too fast that it breaks them

**Reply by a1exhedb** [reply]
Turning off MSync has no effect, same result with a black screen.
2.0.2 with ESync, works as previously stated.

**Reply by ddouble** [reply]
@ohaiibuzzle  your way works for Diablo2 Resurrected on Whisky 2.1.0,  is there an environment variable to turn off **MSync**?

**Reply by larryisthere** [reply]
Diablo IV and Diablo II: Resurrected should be playable using this method. Most cases where they are not playable are due to not saving properly. You can confirm this by making sure to press Enter after entering "1" to save.

**Reply by larryisthere** [reply]
@aquaibm Yes. I've tried both of these two games yesterday. They ran as perfectly as before with these two settings. What was the exact response when you ran D4?

**Reply by ningel** [reply]
I have the same Symptom here. After switching to 2.1 Diablo4 won't run. But the Switch back to 2.0.2 doesn't help :(

**Reply by JSW-Outfit** [reply]
Got Diablo IV working on 2.1.0!

D4 still got stuck in the endless update loop on 2.1.0. with WINEESYNC=1 on all these apps: "Battle.net Launcher.exe", "Battle.net.exe", "Diablo IV Launcher.exe" and "Diablo IV.exe" and disabled MSYNC.

I then went and downloaded Whisky 2.0.2 and ran D4 using the same bottle (and environment variables). After this, I fired back up Whisky 2.1.0, disabled MSYNC (again) and Lo and Behold - Diablo IV runs! It will still occasionally say "Update" in Battle.net but will launch correctly on 2.1.0.

EDIT:
I removed all the environment variables in 2.1.0 and D4 still launches and run. Go figure!

**Reply by aquaibm** [reply]
@larryisthere I just figured out why it's not working by setting up WINEESYNC=1, because I put that under "Diablo IV.exe" instead of "Battle.net Launcher.exe", Once I put that under "Battle.net Launcher.exe", it's working like a charm.

**Reply by ningel** [reply]
Thank you so much! Now all is working fine, right before S2 :)

**Reply by woodymoo** [reply]
Thanks , please add "WINEESYNC=1 " command parameter into wiki for diablo4 for version 2.1.0+.

**Reply by Mhmnz** [reply]
how to run diablo4 through steam? I try many ,but it do not work.

**Comment by altynbek07** [comment]
"(GPTK needs to be re-downloaded)". How can I re-download GPTK? I need delete my current bottle and Whisky?

**Reply by aquaibm** [reply]
Just click "Whisky" menu in the top left corner, choose "Install", it's so simple that I am wondering why do you ask this question?

**Comment by iskandar9ll** [comment]
Does whisky support sim wheels? Please let me know

**Comment by Enigma0901** [comment]
this is probably a stupid question, but is there ever going to be a workaround for EAC?

**Comment by zw0nner** [comment]
maybe someone has solved the problem with this [EAC client cannot be instantiated](https://steamcommunity.com/app/381210/discussions/0/1627412171705899773/) ?

**Comment by DimSar01** [comment]
It can't download the game porting toolkit. Goes incredibly slow on a 200Mbps connection and just stops after it downloads 3.5 MB of it requiring me to force quit the app
